v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/WriteToGCS.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.transforms;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import org.apache.avro.Schema;
+
+/** The {@link WriteToGCS} interface provides all the static values and methods. */
+public interface WriteToGCS {
+
+  /** Set Enum FileFormats for all supported File formats. */
+  enum FileFormats {
+    TEXT,
+    AVRO,
+    PARQUET;
+  }
+
+  EnumMap<FileFormats, String> FILE_FORMATS_MAP =
+      new EnumMap<FileFormats, String>(FileFormats.class);
+
+  /** File Suffix to be set based on Format of the File. */
+  static void setFileFormatSuffix() {
+    FILE_FORMATS_MAP.put(FileFormats.TEXT, ".txt");
+    FILE_FORMATS_MAP.put(FileFormats.AVRO, ".avro");
+    FILE_FORMATS_MAP.put(FileFormats.PARQUET, ".parquet");
+  }
+
+  String TEXT = "TEXT";
+  String AVRO = "AVRO";
+  String PARQUET = "PARQUET";
```

Comment:
This  should be done as part of the String representation for the Enum itself and not separately.